docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,214 @@
+# thin-edge Domain Model
+
+The following section introduces the **thin-edge** domain model.
+
+The **thin-edge** domain model explains details of target domains **thin-edge** is primarily designed for.
+It identifies _entities_ and _aspects_ of those target domains, that are touch points for **thin-edge**.
+Additionally it introduces new _entities_ or _aspects_ to seamlessly fit **thin-edge** into those target domains.
+
+Finally the **thin-edge** domain model gives..
+  * _target domain experts_ a clear idea how to position **thin-edge** in their target domain
+  * _target domain experts_ **and** _thin-edge developers_ a common understanding and unique vocabulary for **thin-edge** and it's environment
+
+### Target Domains
+
+**thin-edge** is designed to facilitate IoT functionality to resource constrained **devices**.
+Therefor it interfaces **devices** with **clouds**, and focuses on both target domains as below:
+  1. Device Domain
+  2. IoT Cloud Domain
+
+Both domains are explained in next sections.
+
+
+## Device Domain
```

Comment:
I would suggest to remove this section "Device Domain". It tries to define terms, which are NOT defined by thin-edge, so they are not part of the thin-edge domain model.

Reply:
Accepted, it is not part of the thin-edge domain model. However it is important to have same understanding and vocabularity for "devices" in context of thin-edge and it's domain model.

So I put the "Device" section for now to the appendix of that document. Could be maybe later also put to a separate document and just referenced in the domain model.